perllib/FixMyStreet/DB/ResultSet/Nearby.pm

```diff
@@ -23,8 +23,17 @@ sub nearby {
         if $args{ids};
     $params->{'problem.category'} = $args{categories} if $args{categories} && @{$args{categories}};
 
-    $params->{$c->stash->{report_age_field}} = { '>=', \"current_timestamp-'$args{report_age}'::interval" }
-        if $args{report_age};
+    my $report_age = $args{report_age};
+    if ( $report_age && ref $report_age eq 'HASH' ) {
+        $params->{'-or'} = FixMyStreet::DB::ResultSet::Problem->report_age_subquery(
+            state_table      => 'problem',
+            report_age       => $report_age,
+            report_age_field => $c->stash->{report_age_field},
+        );
+    } else {
+        $params->{$c->stash->{report_age_field}} = { '>=', \"current_timestamp-'$report_age'::interval" } if
```

Comment:
Now it's reordered like this, I'd probably put the `if $report_age` here instead of the `else`, so it's `elsif ($report_age)`.